src/app/core/services/error.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable, Injector, isDevMode,ErrorHandler } from '@angular/core'
+import { Router } from "@angular/router";
+
+import { from } from 'rxjs';
+
+
+@Injectable({
+    providedIn: 'root'
+})
+export class GlobalErrorHandler implements ErrorHandler {
+  private router:Router;
+  private inject:Injector;
+  constructor(inject:Injector){
+    this.inject = inject;
+  }
+
+  handleError(error) {
+    if (!isDevMode()) {
+      this.router = this.inject.get(Router);
+      this.router.navigate(['error']);
+      console.error(error);    
```

Comment:
You can just console error at the beginning of method hence you avoid “else” statement